File: gateway/src/main/java/com/beat/gateway/GatewayConfigGroup.java
package com.beat.gateway;

import com.beat.gateway.guest.internal.config.GuestAccessConfig;
import com.beat.gateway.refreshtoken.internal.config.RefreshTokenConfig;

public enum GatewayConfigGroup {

	REFRESH_TOKEN_STORE(RefreshTokenConfig.class),
	GUEST_ACCESS(GuestAccessConfig.class);

	private final Class<?> configClass;

	GatewayConfigGroup(Class<?> configClass) {
		this.configClass = configClass;
	}

	public Class<?> getConfigClass() {
		return configClass;
	}
}

File: gateway/src/main/kotlin/com/beat/gateway/GatewayConfigGroup.kt
package com.beat.gateway

import com.beat.gateway.guest.internal.config.GuestAccessConfig
import com.beat.gateway.refreshtoken.internal.config.RefreshTokenConfig

enum class GatewayConfigGroup(
    val configClass: Class<*>,
) {
    REFRESH_TOKEN_STORE(RefreshTokenConfig::class.java),
    GUEST_ACCESS(GuestAccessConfig::class.java),
}
